grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/TaskState.java

```diff
@@ -116,22 +116,22 @@ public static TaskState of(TaskId id) {
     }
 
     public static TaskState of(TaskId id, TaskStatus taskStatus) {
-        return new TaskState(null, null, null, id, null, taskStatus);
+        return new TaskState(null, null, null, id.getValue(), null, taskStatus);
     }
 
     @JsonCreator
     public TaskState(@JsonProperty("taskClass") Class<?> taskClass,
             @JsonProperty("creator") String creator,
             @JsonProperty("schedule") TaskSchedule schedule,
-            @JsonProperty("id") TaskId id,
+            @JsonProperty("id") String id,
```

Comment:
This seems a step backwards! Why do we have to do it?

Reply:
Serialisation issues within RedisQ due to varying representations. In order for Jackson to build the `TaskId` it needs this:

```
"id" : {
    "value" : "blab"
}
```

BUT what we actually use almost everywhere is:

```
"id" : "blab"
```

I.e. we don't use the embedded object just the string literal.

Reply:
Hmm... is this where the `@JsonValue` annotation saves us? Now that I understand what it does :P

Reply:
Update: yeah, I definitely think we can solve this. You just have to update the `TaskId#of` factory to use `@JsonValue` instead of `@JsonProperty`.

Reply:
No `JsonValue` only serialises one way. You would have to write a custom serialiser which kind of defeats the purpose of all this.

Reply:
Just tried it. No go.